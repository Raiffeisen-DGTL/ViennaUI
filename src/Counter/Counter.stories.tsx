import React from 'react';
import { Story, Meta } from 'storybook';
import { Counter, CounterProps } from './Counter';
import { Groups } from '../Groups';

export default {
    title: 'Development/Counter',
    component: Counter,
} as Meta;

export const Overview: Story<CounterProps> = (args) => {
    return (
        <Groups>
            <Counter>8</Counter>
            <Counter>99</Counter>
            <Counter>99+</Counter>
        </Groups>
    );
};

export const CounterStyles: Story<CounterProps> = (args) => {
    return (
        <Groups>
            <Counter design='critical'>9</Counter>
            <Counter design='warning'>12</Counter>
            <Counter design='success'>25</Counter>
            <Counter design='accent'>99+</Counter>
            <Counter design='neutral'>178</Counter>
            <Counter design='disabled'>256</Counter>
        </Groups>
    );
};
