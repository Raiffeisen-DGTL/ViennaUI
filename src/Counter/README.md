# Counter

Компонент отображения количества уведомлений. Используется в навигации второго уровня, табах и в виджетах.

## Импорт

```
import { Counter } from 'vienna-ui';
```

## Свойства / Props


| Prop | Type | Default | Description |
| --- | --- | --- | --- |
| design | CounterDesign \| undefined |  |
| position | "relative" \| "absolute" \| undefined |  |
| top | string \| undefined |  | Свойство для вертикального позиционирования. Зависит от свойства position |
| bottom | string \| undefined |  | Свойство для вертикального позиционирования. Зависит от свойства position |
| left | string \| undefined |  | Свойство для горизонтального позиционирования. Зависит от свойства position |
| right | string \| undefined |  | Свойство для горизонтального позиционирования. Зависит от свойства position |

## Использование

```
    <Groups>
        <Counter>8</Counter>
        <Counter>99</Counter>
        <Counter>99+</Counter>
    </Groups>
```

## Внешний вид

```
    <Groups>
        <Counter design='critical'>9</Counter>
        <Counter design='warning'>12</Counter>
        <Counter design='success'>25</Counter>
        <Counter design='accent'>99+</Counter>
        <Counter design='neutral'>178</Counter>
        <Counter design='disabled'>256</Counter>
    </Groups>
```
