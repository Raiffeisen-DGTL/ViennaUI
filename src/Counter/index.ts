export { Counter, type CounterProps, type CounterDesign } from './Counter';
