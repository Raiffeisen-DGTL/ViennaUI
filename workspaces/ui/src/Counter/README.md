# Counter

Компонент отображения количества уведомлений. Используется в навигации второго уровня, табах и в виджетах.

## Импорт

```
import { Counter } from 'vienna-ui';
```

## Свойства / Props

Свойства наследуются от [HTMLAttributes<HTMLDivElement>](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L1746)

Prop | Type | Default | Description
--- | --- | --- | ---
design | "critical" \| "warning" \| "success" \| "accent" \| "neutral" \| "disabled" \| undefined | false |
position | "relative" \| "absolute" \| undefined | false |
top | string \| undefined | false | Свойство для вертикального позиционирования. Зависит от свойства position
bottom | string \| undefined | false | Свойство для вертикального позиционирования. Зависит от свойства position
left | string \| undefined | false | Свойство для горизонтального позиционирования. Зависит от свойства position
right | string \| undefined | false | Свойство для горизонтального позиционирования. Зависит от свойства position

## Использование

```jsx
<Counter>99+</Counter>
```
