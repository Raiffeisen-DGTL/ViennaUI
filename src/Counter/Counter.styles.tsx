import styled, { css } from 'styled-components';
import { getPresets } from '../Utils/styling';
import { CounterDesign } from './Counter';

const presets = getPresets('counter', {
    base: null,
    design: '$design',
});

export interface PropsBox {
    $design?: CounterDesign;
    $position?: 'relative' | 'absolute';
    $top?: string;
    $bottom?: string;
    $left?: string;
    $right?: string;
}
export const Box = styled.span<PropsBox>`
    position: ${({ $position }) => $position};
    display: inline-flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    box-sizing: border-box;

    ${presets.base}
    ${presets.design}

    ${({ $position, $top, $bottom, $left, $right }) =>
        $position === 'absolute' &&
        css`
            top: ${$top ?? 'auto'};
            bottom: ${$bottom ?? 'auto'};
            left: ${$left ?? 'auto'};
            right: ${$right ?? 'auto'};
        `}
`;
