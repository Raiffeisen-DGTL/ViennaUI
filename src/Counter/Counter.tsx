import React, { FC } from 'react';
import { Box, PropsBox } from './Counter.styles';
import { BoxStyled } from '../Utils/styled';

export type CounterDesign = 'critical' | 'warning' | 'success' | 'accent' | 'neutral' | 'disabled';

export interface CounterProps extends BoxStyled<typeof Box, PropsBox> {
    design?: CounterDesign;
    position?: 'relative' | 'absolute';
    top?: string;
    bottom?: string;
    left?: string;
    right?: string;
}

export const Counter: FC<CounterProps> = ({
    design = 'neutral',
    position = 'relative',
    top,
    bottom,
    left,
    right,
    ...attrs
}) => {
    return (
        <Box {...attrs} $design={design} $position={position} $top={top} $bottom={bottom} $left={left} $right={right} />
    );
};

Counter.displayName = 'Counter';
