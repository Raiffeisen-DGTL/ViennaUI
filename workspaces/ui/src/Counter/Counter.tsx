import React, { HTMLAttributes } from 'react';
import { Box } from './Counter.styles';

export interface CounterProps extends HTMLAttributes<HTMLDivElement> {
    children?: React.ReactNode;
    design?: 'critical' | 'warning' | 'success' | 'accent' | 'neutral' | 'disabled';
    position?: 'relative' | 'absolute';
    top?: string;
    bottom?: string;
    left?: string;
    right?: string;
}

export const Counter: React.FC<CounterProps> = (props): JSX.Element => {
    const { children, ...attrs } = props;
    return <Box {...attrs}>{children}</Box>;
};

Counter.defaultProps = {
    design: 'neutral',
    position: 'relative',
};

Counter.displayName = 'Counter';
export default Counter;
