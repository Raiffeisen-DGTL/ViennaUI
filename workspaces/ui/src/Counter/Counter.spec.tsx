import React from 'react';
import Counter, { CounterProps } from './Counter';

test('Counter', () => {
    const snap = snapshot.render(<Counter>18</Counter>);
    expect(snap).toMatchSnapshot();
});

test('Counter w/ design', () => {
    const designs: CounterProps['design'][] = ['critical', 'warning', 'success', 'accent', 'neutral', 'disabled'];

    designs.forEach((design) => {
        const snap = snapshot.render(<Counter design={design}>18</Counter>);
        expect(snap).toMatchSnapshot(design);
    });
});

test('Counter w/ position', () => {
    const positions: CounterProps['position'][] = ['relative', 'absolute'];

    positions.forEach((position) => {
        const snap = snapshot.render(
            <Counter position={position} top='10px' bottom='10px' left='10px' right='10px'>
                18
            </Counter>
        );

        expect(snap).toMatchSnapshot(position);
    });
});

test('Counter w/ any props', () => {
    const props = { className: 'className', id: 'Counter', title: 'title', onClick: () => 'Counter' };
    const component = <Counter {...props}>18</Counter>;

    const snap = snapshot.render(component);
    expect(snap).toMatchSnapshot();

    const wrapper = snapshot.shallow(component);
    for (const [key, value] of Object.entries(props)) {
        expect(wrapper.props()[key]).toEqual(value);
    }
});
