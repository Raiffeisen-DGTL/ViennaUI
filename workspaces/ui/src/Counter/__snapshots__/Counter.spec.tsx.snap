// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter 1`] = `
<span
  class="Counterstyles__Box-oocehd-0 eQsGEI"
>
  18
</span>
`;

exports[`Counter w/ any props 1`] = `
<span
  class="Counterstyles__Box-oocehd-0 eQsGEI className"
  id="Counter"
  title="title"
>
  18
</span>
`;

exports[`Counter w/ design: accent 1`] = `
<span
  class="Counterstyles__Box-oocehd-0 cCaivE"
>
  18
</span>
`;

exports[`Counter w/ design: critical 1`] = `
<span
  class="Counterstyles__Box-oocehd-0 dBmyGU"
>
  18
</span>
`;

exports[`Counter w/ design: disabled 1`] = `
<span
  class="Counterstyles__Box-oocehd-0 iVMrQX"
>
  18
</span>
`;

exports[`Counter w/ design: neutral 1`] = `
<span
  class="Counterstyles__Box-oocehd-0 eQsGEI"
>
  18
</span>
`;

exports[`Counter w/ design: success 1`] = `
<span
  class="Counterstyles__Box-oocehd-0 igFKyr"
>
  18
</span>
`;

exports[`Counter w/ design: warning 1`] = `
<span
  class="Counterstyles__Box-oocehd-0 kXEOzk"
>
  18
</span>
`;

exports[`Counter w/ position: absolute 1`] = `
<span
  class="Counterstyles__Box-oocehd-0 lmpgYJ"
>
  18
</span>
`;

exports[`Counter w/ position: relative 1`] = `
<span
  class="Counterstyles__Box-oocehd-0 eQsGEI"
>
  18
</span>
`;
